motor_control/throttle/base_throttle.ino

```diff
@@ -1,15 +1,54 @@
 #include "base_throttle.h"
 
-Throttle::Throttle(const int &pin)
+Throttle::Throttle()
 {
-    m_sensor_pin = pin;
-    m_raw_value = 0;
-    m_motor_value = 0;
+    lastMotorWrite = 0;
+}
+
+
+void Throttle::run(){
+    setup();
+    loop();
+}
+
+void Throttle::setup() {
+  // Initialisation of ESC
+  ESC.attach(ESCOutput, minPulseWidth, maxPulseWidth);
+}
+
+void Throttle::writeToESC(int input_pin) { 
+  ESC.write(map(input_pin, 0, 1023, 0, 180));
+}
```

Comment:
Like we mentioned on the call, we are not writing to any ESC or motor controller within the throttle class. Your responsibility is simply to make sure that `m_motor_value` is updated when `compute_motor_value()` is called.